sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -558,3 +561,42 @@ AZ_NODISCARD az_result az_iot_provisioning_client_parse_received_topic_and_paylo
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_provisioning_client_get_request_payload(
+    az_iot_provisioning_client const* client,
+    az_span custom_payload_property,
+    az_iot_provisioning_client_payload_options const* reserved,
+    uint8_t* mqtt_payload,
+    size_t mqtt_payload_size,
+    size_t* out_mqtt_payload_length)
+{
+  (void)reserved;
+
+  _az_PRECONDITION_NOT_NULL(client);
+  _az_PRECONDITION_IS_NULL(reserved);
```

Comment:
It's probably fine, but are we sure we want this check? How likely is it we'll add options in the future?

Reply:
It's possible but fairly unlikely we'd add options.  The most likely scenario would be we wanted app to specify some custom JSON into the {...} top level payload.  The issue though is that the `custom_payload` already is the escape hatch for apps to add any arbitrary JSON they want, in a payload:<userDefined>.  So putting yet another option as a peer to "payload" feels unlikely.  Though I'm all for having an escape hatch.

I don't really have strong opinions either way on this as most devs know to leave something like this as NULL.